File: src/main/java/eterea/api/rest/model/Articulo.java
/**
 * Entidad Articulo vinculada a la tabla articulos, a us vez esta tabla esta relacionada con
 * articulofecha, productoarticulo y articulomovimientoprevio
 *
 */
package eterea.api.rest.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.math.BigDecimal;

@Data
@Entity
@Table(name = "articulos")
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper=false)
public class Articulo extends Auditable implements Serializable {

	private static final long serialVersionUID = 3993609223567470857L;

	@Id
	@Column(name = "codigo")
	private String articuloId;

	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "clave")
	private Long uniqueId;

	@Column(name = "art_neg_id")
	private Integer negocioId;

	private String descripcion = "";

	@Column(name = "art_voucher")
	private String leyendaVoucher = "";

	@Column(name = "precioventasiniva")
	private BigDecimal precioVentaSinIva = BigDecimal.ZERO;

	@Column(name = "preciounitario")
	private BigDecimal precioVentaConIva = BigDecimal.ZERO;

	@Column(name = "cgocontable")
	private Long numeroCuentaVentas;

	@Column(name = "cgocontablecompras")
	private Long numeroCuentaCompras;

	@Column(name = "Cgocontablegastos")
	private Long numeroCuentaGastos;

	@Column(name = "cgocentro")
	private Integer centroStockId;

	@Column(name = "cgorubro")
	private Long rubroId;

	@Column(name = "cgosubrubro")
	private Long subRubroId;

	@Column(name = "preciocompra")
	private BigDecimal precioCompra = BigDecimal.ZERO;

	@Column(name = "iva105")
	private Byte iva105 = 0;

	@Column(name = "art_compraneto")
	private BigDecimal precioCompraNeto = BigDecimal.ZERO;

	private Byte exento = 0;

	@Column(name = "stockminimo")
	private Long stocMinimo = 0L;

	@Column(name = "stockoptimo")
	private Long stockOptimo = 0L;

	@Column(name = "bloqueocompras")
	private Byte bloqueoCompras = 0;

	@Column(name = "bloqueostock")
	private Byte bloqueoStock = 0;

	@Column(name = "bloqueoventas")
	private Byte bloqueoVentas = 0;

	@Column(name = "art_ume_id")
	private Long unidadMedidaId;

	@Column(name = "art_condecimales")
	private Byte conDecimales = 0;

	@Column(name = "art_ventas")
	private Byte ventas = 0;

	@Column(name = "art_compras")
	private Byte compras = 0;

	@Column(name = "umedida")
	private String unidadMedida = "";

	@Column(name = "art_con_id")
	private Integer conversionId;

	@Column(name = "art_ventasinstock")
	private Byte ventaSinStock = 0;

	@Column(name = "art_controlstock")
	private Byte controlaStock = 0;

	@Column(name = "art_asientocostos")
	private Byte asientoCostos = 0;

	@Column(name = "art_maskbal")
	private String mascaraBalanza = "";

	@Column(name = "art_habingreso")
	private Byte habilitaIngreso = 0;

	@Column(name = "art_comision")
	private BigDecimal comision = BigDecimal.ZERO;

	@Column(name = "art_pre_ID")
	private Integer prestadorId;

}

File: src/main/kotlin/eterea/api/rest/model/Articulo.kt
package eterea.api.rest.model

import jakarta.persistence.Entity
import jakarta.persistence.Table
import jakarta.persistence.Id
import jakarta.persistence.Column
import java.math.BigDecimal
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType

@Entity
@Table(name = "articulos")
data class Articulo(

	@Id
	@Column(name = "codigo")
	var articuloId: String = "",

	@Column(name = "art_neg_id")
	var negocioId: Int = 0,

	var descripcion: String = "",

	@Column(name = "art_voucher")
	var leyendaVoucher: String = "",

	@Column(name = "precioventasiniva")
	var precioVentaSinIva: BigDecimal = BigDecimal.ZERO,

	@Column(name = "preciounitario")
	var precioVentaConIva: BigDecimal = BigDecimal.ZERO,

	@Column(name = "cgocontable")
	var cuentaVentas: Long? = null,

	@Column(name = "cgocontablecompras")
	var cuentaCompras: Long? = null,

	@Column(name = "Cgocontablegastos")
	var cuentaGastos: Long? = null,

	@Column(name = "cgocentro")
	var centroStockId: Int? = null,

	@Column(name = "cgorubro")
	var rubroId: Long? = null,

	@Column(name = "cgosubrubro")
	var subRubroId: Long? = null,

	@Column(name = "preciocompra")
	var precioCompra: BigDecimal = BigDecimal.ZERO,

	@Column(name = "iva105")
	var iva105: Byte = 0,

	@Column(name = "art_compraneto")
	var precioCompraNeto: BigDecimal = BigDecimal.ZERO,

	@Column(name = "exento")
	var exento: Byte = 0,

	@Column(name = "stockminimo")
	var stocMinimo: Long = 0L,

	@Column(name = "stockoptimo")
	var stockOptimo: Long = 0L,

	@Column(name = "bloqueocompras")
	var bloqueoCompras: Byte = 0,

	@Column(name = "bloqueostock")
	var bloqueoStock: Byte = 0,

	@Column(name = "bloqueoventas")
	var bloqueoVentas: Byte = 0,

	@Column(name = "art_ume_id")
	var unidadMedidaId: Long? = null,

	@Column(name = "art_condecimales")
	var conDecimales: Byte = 0,

	@Column(name = "art_ventas")
	var ventas: Byte = 0,

	@Column(name = "art_compras")
	var compras: Byte = 0,

	@Column(name = "umedida")
	var unidadMedida: String = "",

	@Column(name = "art_con_id")
	var conversionId: Int? = null,

	@Column(name = "art_ventasinstock")
	var ventaSinStock: Byte = 0,

	@Column(name = "art_controlstock")
	var controlaStock: Byte = 0,

	@Column(name = "art_asientocostos")
	var asientoCostos: Byte = 0,

	@Column(name = "art_maskbal")
	var mascaraBalanza: String = "",

	@Column(name = "art_habingreso")
	var habilitaIngreso: Byte = 0,

	@Column(name = "art_comision")
	var comision: BigDecimal = BigDecimal.ZERO,

	@Column(name = "art_pre_ID")
	var prestadorId: Int? = null,

	@Column(name = "clave")
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	var autoNumericoId: Long? = null

) : Auditable()

